osfoffline/views/start_screen.py

```diff
@@ -21,47 +32,67 @@ def __init__(self):
         super().__init__()
         self.start_screen = Ui_startscreen()
 
+    @asyncio.coroutine
     def log_in(self):
-        user_name = self.start_screen.emailEdit.text().strip()
-        password = self.start_screen.passwordEdit.text().strip()
-
-        # assumption: logged in user properly.
-        # assumption: all the fields needed for db are set
-        full_name = user_name
-        osf_id = user_name
-        oauth_token = password
-        # check if user already exists in db
-
+        logging.debug('attempting to log in')
+        personal_access_token = self.start_screen.personalAccessTokenEdit.text().strip()
+        url = settings.API_BASE + '/v2/users/me/'
+        loop = asyncio.get_event_loop()
+        osf = OSFQuery(loop, personal_access_token)
+        logging.debug('url: {}'.format(url))
         try:
-            user = session.query(User).filter(User.osf_login == user_name).one()
-            user.logged_in = True
-            save(session, user)
-            self.close()
-        except MultipleResultsFound:
-            logging.warning(
-                'multiple users with same username. deleting all users with this username. restarting function.')
-            for user in session.query(User):
-                if user.osf_login == user_name:
+            resp = yield from osf.make_request(url, expects=[200])
+        except (aiohttp.errors.ClientTimeoutError, aiohttp.errors.ClientConnectionError, aiohttp.errors.TimeoutError):
+            # No internet connection
+            QMessageBox.warning(
+                None,
+                "Log in Failed",
+                "Unable to connect to server. Check your internet connection or try again later."
+            )
+        except aiohttp.errors.HttpBadRequest:
+            # Invalid credentials
+            QMessageBox.warning(
+                None,
+                "Log in Failed",
+                "Invalid login credentials."
+            )
+        else:
+            json_resp = yield from resp.json()
+            remote_user = RemoteUser(json_resp['data'])
+            full_name = json_resp['data']['attributes']['full_name']     # Use one of these later to "Welcome {user}"
```

Comment:
This can be changed to `remote_user.name`. If you're going to use an abstracted data parsing class, use it consistently.